src/can.rs

```diff
@@ -0,0 +1,272 @@
+//! CAN mock implementations.
+//!
+//! ## Usage
+//!
+//! ```
+//! extern crate embedded_hal;
+//! extern crate embedded_hal_mock;
+//!
+//! use embedded_can::blocking::Can;
+//! use embedded_can::Frame;
+//! use embedded_can::{StandardId, ExtendedId};
+//! use embedded_hal_mock::can::{Mock as CanMock, Transaction as CanTransaction, MockFrame as CanMockFrame};
+//!
+//! // Create CAN frame
+//!
+//! let id: StandardId = StandardId::new(0x123).unwrap();
+//! let frame1 = CanMockFrame::new(id, &[0x00, 0x00, 0x00, 0x00, 0x10]).unwrap();
+//! let id: ExtendedId = ExtendedId::new(0x12345678).unwrap();
+//! let frame2 = CanMockFrame::new(id, &[0x00, 0x00, 0x00, 0x00, 0x10]).unwrap();
+//!
+//! // Configure expectations
+//! let expectations = [
+//!     CanTransaction::transmit(frame1.clone()),
+//!     CanTransaction::receive(frame2.clone()),
+//! ];
+//!
+//! let mut can = CanMock::new(&expectations);
+//!
+//! // Transmitting
+//! embedded_can::blocking::Can::transmit(&mut can, &frame1.clone()).unwrap();
+//!
+//! // Receiving
+//! let frame = embedded_can::blocking::Can::receive(&mut can).unwrap();
+//! assert_eq!(frame, frame2.clone());
+//!
+//! // Finalise expectations
+//! can.done();
+//! ```
+//!
+use embedded_can::{ErrorKind, Frame, Id};
+use embedded_hal_nb::nb;
+
+use crate::common::Generic;
+use crate::error::MockError;
+
+/// CAN Transaction modes
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Mode {
+    /// Transmit transaction
+    Transmit,
+    /// Receive transaction
+    Receive,
+}
+
+/// CAN Transaction type
+///
+/// Models an CAN transmit or receive transaction
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Transaction {
+    expected_mode: Mode,
+    expected_frame: Option<MockFrame>,
```

Comment:
because it's either rx or tx there's only one frame (unlike SPI where you have both incoming and outgoing data) so it seems like `expected_frame` and `response_frame` could just be `frame`?